### PR TITLE
Add test for ensuring migration numbers are up and to the right

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes https://github.com/google/exposure-notifications-verification-server/issues/1343

**Release Note**

```release-note
Add test for ensuring database migration numbers are strictly increasing
```

### CODE REVIEW

File: pkg/database/database_util.go [hunk 1]
```diff
@@ -343,7 +343,7 @@ func runMigrations(db *Database) error {
 	}()
 
 	// Run migrations.
-	if err := db.RunMigrations(ctx); err != nil {
+	if err := db.MigrateTo(ctx, "", false); err != nil {
```
Comment (whaught) [inline pkg/database/database_util.go:346]: Aren't these equivalent? Why the change

Comment (sethvargo) [reply pkg/database/database_util.go:346]: Yea, that's why I dropped `.RunMigrations` 😄

Comment (sethvargo) [reply pkg/database/database_util.go:346]: We had two functions that did the same thing.

**Review by whaught — APPROVED** [review]
